internal/testdata/testing_bcksoutput.golden

```diff
@@ -0,0 +1,15 @@
+package state
+
+// Code generated by weld. DO NOT EDIT.
+
+import (
+	"testing"
+)
+
+type Backends interface {
+	GetFoo() Foo
+
+	// Setters that allow setting the implementation of the above getter for a single test case. On t.Cleanup
+	// the dependency is reset to it's original value / implementation.
+	SetGetFooForTesting(t *testing.T, foo Foo)
```

Comment:
Rather than having setters on the Backends I was thinking you would rather have another function that can be called to wrap the backends in some overridden dependencies.

So your calls would look like
```
b, err := makeBackends()
if err {...}
b = overrideForTesting(b, WithDB(...), WithFoo(...))
```
so we would need to generate a function that can match that second signature.

Reply:
I feel quite strongly that an independent testing state can be generated. I also like the ability to specify an override per testing.T session.

Do you feel strongly about it being a wrapper of a specific backends and not it's own?

Reply:
Yes, keep it separate, you can achieve both of your goals with a separate struct